src/lib.rs

```diff
@@ -197,7 +259,7 @@ pub(crate) struct WriteHalf {
   vectored: bool,
   auto_apply_mask: bool,
   writev_threshold: usize,
-  write_buffer: Vec<u8>,
+  buffer: BytesMut,
```

Comment:
Using Vec is faster in my testing. To replicate `buffer.advance(written)` you can do `buffer.splice(..written, [0u8; 0])`. Because this is expensive what I found is better is to instead have a `buf_pos: usize` that I increment, and only in `start_send` do I run
```rust
self.buffer.splice(..self.buf_pos, [0u8; 0]);
self.buf_pos = 0;
```
before the fmt_head call to write the frame into the buffer. This is because `BytesMut::advance` is kinda expensive it seems.